plugin/plugin.go

```diff
@@ -317,6 +320,47 @@ func wrap_write_callback(ds *C.data_set_t, cvl *C.value_list_t, ud *C.user_data_
 	return 0
 }
 
+// First declare some types, interfaces, general functions
+
+// Shutters are objects that when called will shut down the plugin gracefully
+type Shutter interface {
+	Shutdown() error
+}
+
+// shutdownFuncs holds references to all shutdown callbacks
+var shutdownFuncs = make(map[string]Shutter)
+
+//export wrap_shutdown_callback
+func wrap_shutdown_callback() C.int {
+	if len(shutdownFuncs) <= 0 {
+		return -1
```

Comment:
This should return zero instead of –1: A non-zero return value signals failure, but not having any shutdown callbacks is not an error condition.